netplan/cli/utils.py

```diff
@@ -55,6 +57,19 @@ def nm_running():  # pragma: nocover (covered in autopkgtest)
         return False
 
 
+def nm_interfaces(paths=glob.glob('/run/NetworkManager/system-connections/netplan-*')):
```

Comment:
Ok, so I don't think the default value is correct here. I'd recommend not setting a default value at all.

As it is right now is actually a tricky common Python mistake in most cases! Default values are evaluated and stored once on start of the Python code and never re-ran. This means that the default value of 'paths' will have the state of the `/run/NetworkManager/system-connections/` directory from one particular time, e.g. it will not be reevaluated with consecutive nm_interfaces() calls within the application's life cycle.

This is why I think we should just not have a default value - we don't use that anyway!

Reply:
Ahh, very good catch!
Let's get rid of that default. As you said, we're not using it after all.

Fixed.